.github/workflows/workflow.yml

```diff
@@ -0,0 +1,71 @@
+name: Workflow CI
+
+on:
+  workflow_run:
+    workflows:
+      - "Pull Request CI"
+    types:
+      - completed
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2.3.1
+        with:
+          persist-credentials: false
+      - run: |
+          PR=$(curl https://api.github.com/search/issues?q=${{ github.event.workflow_run.head_sha }} |
+          grep -Po "(?<=${{ github.event.workflow_run.repository.full_name }}\/pulls\/)\d*" | head -1)
+          echo "PR=$PR" >> $GITHUB_ENV
+      - run: |
+          git fetch origin ${{ github.event.workflow_run.head_sha }}
+          git checkout ${{ github.event.workflow_run.head_sha }}
```

Comment:
Doesn't this job still run code (`npm test`) from an untrusted head_sha, with the secrets in scope? Not sure if I'm right about that.

Is there any way to get the build results from "Pull Request CI", which were built without secrets in scope, and upload them from this workflow which does have secrets in scope?

Reply:
@kainino0x right that's possible, I will update this PR to use the build from base Action